nomos-services/api/src/http/mempool.rs

```diff
@@ -36,7 +36,8 @@ where
         + Clone
         + 'static,
     StorageAdapter::Error: Debug,
-    Item: Clone + Debug + Send + Sync + Serialize + for<'de> Deserialize<'de> + 'static,
+    Item:
+        Transaction + Clone + Debug + Send + Sync + Serialize + for<'de> Deserialize<'de> + 'static,
     Key: Clone + Debug + Ord + Hash + Send + Sync + Serialize + for<'de> Deserialize<'de> + 'static,
```

Comment:
nit: for<'de> Deserialize<'de> is identical to `DeserializedOwned` 

(I know it's not from your change but I think it's less syntax noise)